apps/api/src/app/environments-v1/usecases/construct-framework-workflow/construct-framework-workflow.usecase.ts

```diff
@@ -178,4 +174,13 @@ export class ConstructFrameworkWorkflow {
       skip: (controlValues) => false,
     };
   }
+  private async getDbWorkflow(environmentId: string, workflowId: string): Promise<NotificationTemplateEntity> {
+    const foundWorkflow = await this.workflowsRepository.findByTriggerIdentifier(environmentId, workflowId);
+
+    if (!foundWorkflow) {
+      throw new InternalServerErrorException(`Workflow ${workflowId} not found`);
```

Comment:
Should this be a NotFound exception? Or just return null?

Reply:
It's an internal error that has nothing to do with the user request, it shows a problem with our internal logic, a 400 refferes to a customer error, it's not the case